---
title: "workshop2020"
bibliography: "`r system.file('vignettes', 'bibliography.bib', package = 'BuildABiocWorkshop2020')`"
csl: "`r system.file('vignettes', 'nature.csl', package = 'BuildABiocWorkshop2020')`"
vignette: >
  %\VignetteIndexEntry{BuildABiocWorkshop2020}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
output:
  bookdown::html_document2:
    theme: simplex
    toc: true
    toc_float: true
    toc_depth: 4
    fig_caption: true
---

```{r checkplatform, include=FALSE}
library(knitr)
library(Rcpp)
opts_chunk$set(eval=TRUE, fig.width=6, fig.height=2,
               warning = FALSE, message = FALSE)
```

## Introduction

It is a routine way to show the distribution of mutation for genetic variations by lollipop-style (or needle-style) plot in a genome browser, along with a variety of genomic annotations, such as gene level or exon level models, CpG island, and so on. To show the SNP status or methylation data, a special plot, called lollipop plot or needle plot, is used to show the distribution along annotations. Many of currently available tools offline or online provide lollipop-like plot to show the mutation data such as cBioPortal Tools::MutationMapper[@gao2013integrative], EMBL-EBI::Pfam[@finn2006pfam], and BioJS::muts-needle-plot[@michael_p_schroeder_2015_14561], BiQ Analyzer[@bock2005biq], and Methylation plotter[@mallona2014methylation]. The cBioPortal Tools::MutationMapper is a well-known and easy to use on-line genome browser that can output high quality of figures with mutations by input tab-delimited mutation data. In R/Bioconductor, there are options to use the flexibility of the Rgraphics system to display Methylation data such as _MethVisual_[@zackay2010methvisual], _REDseq_[@zhu4redseq], and _GenVisR_[@skidmore2016genvisr].
