---
title: "trackViewer_workshop2020"
bibliography: "`r system.file('vignettes', 'bibliography.bib', package = 'test4')`"
csl: "`r system.file('vignettes', 'nature.csl', package = 'test4')`"
vignette: >
  %\VignetteIndexEntry{test4}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
output: 
  bookdown::html_document2:
    theme: simplex
    toc: true
    toc_float: true
    toc_depth: 4
    fig_caption: true
---

```{r checkplatform, include=FALSE, eval=FALSE}
library(knitr)
suppressPackageStartupMessages({
  library(trackViewer)
  library(pander)
  library(VariantAnnotation)
  library(TxDb.Hsapiens.UCSC.hg19.knownGene)
  library(org.Hs.eg.db)
  library(GenomicFeatures)
  library(org.Mm.eg.db)
})
opts_chunk$set(eval=TRUE, fig.width=6, fig.height=2, 
               warning = FALSE, message = FALSE)
```

## Introduction

It is a routine way to show the distribution of mutation for genetic variations by lollipop-style (or needle-style) plot in a genome browser, along with a variety of genomic annotations, such as gene level or exon level models, CpG island, and so on. To show the SNP status or methylation data, a special plot, called lollipop plot or needle plot, is used to show the distribution along annotations. Many of currently available tools offline or online provide lollipop-like plot to show the mutation data such as cBioPortal Tools::MutationMapper[@gao2013integrative], EMBL-EBI::Pfam[@finn2006pfam], and BioJS::muts-needle-plot[@michael_p_schroeder_2015_14561], BiQ Analyzer[@bock2005biq], and Methylation plotter[@mallona2014methylation]. The cBioPortal Tools::MutationMapper is a well-known and easy to use on-line genome browser that can output high quality of figures with mutations by input tab-delimited mutation data. In R/Bioconductor, there are options to use the flexibility of the Rgraphics system to display Methylation data such as _MethVisual_[@zackay2010methvisual], _REDseq_[@zhu4redseq], and _GenVisR_[@skidmore2016genvisr]. 

**Table 1: Tools availble for lollipop plot. **
```{r echo=FALSE}
software <- c("MutationMapper[@gao2013integrative]", "Pfam[@finn2006pfam]", "muts-needle-plot[@michael_p_schroeder_2015_14561]", "BiQ Analyzer[@bock2005biq]", "Methylation plotter[@mallona2014methylation]", "_MethVisual_[@zackay2010methvisual]", "_REDseq_[@zhu4redseq]", "_GenVisR_[@skidmore2016genvisr]")
inputs <- c("tab-delimited text", "JSON", "JSON", "BiQ methylation file", "tab-delimited text", "R list", "R list", "R dataframe")
online <- c("Yes", "Yes", "No", "Yes", "Yes", "No", "No", "No")
description <- c("interprets mutations with different heights along protein annotations in automatic color theme", "could combine different line and head colors with different drawing styles", "plots data point with different colors, heights, and size along annotations, and highlighted selcted coordinates", "interprets methylation status in black & white", "stacked multiple methylation status for multiple samples", "visualize the methylation status of CpGs according to their genomic position", "plot frequencies of methylations and SNPs along a chromosome", "plot most accurate graphic representation of the ensembl annotation version based on biomart service")
df <- data.frame(software=software, inputs=inputs, online=online, description=description)

library(pander)
panderOptions('table.alignment.default', function(df)
    ifelse(sapply(df, is.numeric), 'right', 'left'))
panderOptions('table.split.table', Inf)
panderOptions('keep.trailing.zeros', TRUE)
pander(df)
```


All the tools available for the genomic data visualization mentioned above can meet the basic even complicated requirement. However, if there are multiple mutations in almost same position, it is very difficult to display the data using any available tools. What's more there is a tendency that the figures it generated become more and more complex and busy. Thereby make it difficult in generating high quality pictures for publication in bunch. To fill this gap, we developed _trackViewer_, a R/Bioconductor package as an enhanced light-weight genome viewer for visualizing various types of high-throughput sequencing data, side by side or stacked with multiple annotation tracks. Besides the regular read coverage tracks supported by existing genome browsers/viewers,  _trackViewer_ can also be used to generate lollipop plot to depict the methylation and SNP/mutation status, together with coverage data and annotation tracks to facilitate integrated analysis of multi-omics data. In addition, figures generated by _trackViewer_ are interactive, i.e., the feel-and-look such as the layout, the color scheme and the order of tracks can be easily customized by the users. Furthermore, _trackViewer_ can be easily integrated into standard analysis pipeline for various high-throughput sequencing dataset such as ChIP-seq, RNA-seq, methylation-seq or DNA-seq. The images produced by _trackViewer_ are highly customizable including labels, symbols, colors and size. Here, we illustrate its utilities and capabilities in deriving biological insights from multi-omics dataset from GEO/Encode.
