<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>trackViewer_workshop2020 • test4</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="trackViewer_workshop2020">
<meta property="og:description" content="test4">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">test4</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/trackViewer.html">trackViewer_workshop2020</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/seandavi/BuildABiocWorkshop2020fff">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>trackViewer_workshop2020</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/hukai916/test4fff/blob/master/vignettes/trackViewer.Rmd"><code>vignettes/trackViewer.Rmd</code></a></small>
      <div class="hidden name"><code>trackViewer.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>It is a routine way to show the distribution of mutation for genetic variations by lollipop-style (or needle-style) plot in a genome browser, along with a variety of genomic annotations, such as gene level or exon level models, CpG island, and so on. To show the SNP status or methylation data, a special plot, called lollipop plot or needle plot, is used to show the distribution along annotations. Many of currently available tools offline or online provide lollipop-like plot to show the mutation data such as cBioPortal Tools::MutationMapper<span class="citation"><sup>1</sup></span>, EMBL-EBI::Pfam<span class="citation"><sup>2</sup></span>, and BioJS::muts-needle-plot<span class="citation"><sup>3</sup></span>, BiQ Analyzer<span class="citation"><sup>4</sup></span>, and Methylation plotter<span class="citation"><sup>5</sup></span>. The cBioPortal Tools::MutationMapper is a well-known and easy to use on-line genome browser that can output high quality of figures with mutations by input tab-delimited mutation data. In R/Bioconductor, there are options to use the flexibility of the Rgraphics system to display Methylation data such as <em>MethVisual</em><span class="citation"><sup>6</sup></span>, <em>REDseq</em><span class="citation"><sup>7</sup></span>, and <em>GenVisR</em><span class="citation"><sup>8</sup></span>.</p>
<p><strong>Table 1: Tools availble for lollipop plot. </strong></p>
<pre><code>## Warning: package 'pander' was built under R version 4.0.2</code></pre>
<table class="table">
<colgroup>
<col width="44%">
<col width="19%">
<col width="7%">
<col width="28%">
</colgroup>
<thead><tr class="header">
<th align="left">software</th>
<th align="left">inputs</th>
<th align="left">online</th>
<th align="left">description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">MutationMapper<span class="citation"><sup>1</sup></span>
</td>
<td align="left">tab-delimited text</td>
<td align="left">Yes</td>
<td align="left">interprets mutations with different heights along protein annotations in automatic color theme</td>
</tr>
<tr class="even">
<td align="left">Pfam<span class="citation"><sup>2</sup></span>
</td>
<td align="left">JSON</td>
<td align="left">Yes</td>
<td align="left">could combine different line and head colors with different drawing styles</td>
</tr>
<tr class="odd">
<td align="left">muts-needle-plot<span class="citation"><sup>3</sup></span>
</td>
<td align="left">JSON</td>
<td align="left">No</td>
<td align="left">plots data point with different colors, heights, and size along annotations, and highlighted selcted coordinates</td>
</tr>
<tr class="even">
<td align="left">BiQ Analyzer<span class="citation"><sup>4</sup></span>
</td>
<td align="left">BiQ methylation file</td>
<td align="left">Yes</td>
<td align="left">interprets methylation status in black &amp; white</td>
</tr>
<tr class="odd">
<td align="left">Methylation plotter<span class="citation"><sup>5</sup></span>
</td>
<td align="left">tab-delimited text</td>
<td align="left">Yes</td>
<td align="left">stacked multiple methylation status for multiple samples</td>
</tr>
<tr class="even">
<td align="left">
<em>MethVisual</em><span class="citation"><sup>6</sup></span>
</td>
<td align="left">R list</td>
<td align="left">No</td>
<td align="left">visualize the methylation status of CpGs according to their genomic position</td>
</tr>
<tr class="odd">
<td align="left">
<em>REDseq</em><span class="citation"><sup>7</sup></span>
</td>
<td align="left">R list</td>
<td align="left">No</td>
<td align="left">plot frequencies of methylations and SNPs along a chromosome</td>
</tr>
<tr class="even">
<td align="left">
<em>GenVisR</em><span class="citation"><sup>8</sup></span>
</td>
<td align="left">R dataframe</td>
<td align="left">No</td>
<td align="left">plot most accurate graphic representation of the ensembl annotation version based on biomart service</td>
</tr>
</tbody>
</table>
<p>All the tools available for the genomic data visualization mentioned above can meet the basic even complicated requirement. However, if there are multiple mutations in almost same position, it is very difficult to display the data using any available tools. What’s more there is a tendency that the figures it generated become more and more complex and busy. Thereby make it difficult in generating high quality pictures for publication in bunch. To fill this gap, we developed <em>trackViewer</em>, a R/Bioconductor package as an enhanced light-weight genome viewer for visualizing various types of high-throughput sequencing data, side by side or stacked with multiple annotation tracks. Besides the regular read coverage tracks supported by existing genome browsers/viewers, <em>trackViewer</em> can also be used to generate lollipop plot to depict the methylation and SNP/mutation status, together with coverage data and annotation tracks to facilitate integrated analysis of multi-omics data. In addition, figures generated by <em>trackViewer</em> are interactive, i.e., the feel-and-look such as the layout, the color scheme and the order of tracks can be easily customized by the users. Furthermore, <em>trackViewer</em> can be easily integrated into standard analysis pipeline for various high-throughput sequencing dataset such as ChIP-seq, RNA-seq, methylation-seq or DNA-seq. The images produced by <em>trackViewer</em> are highly customizable including labels, symbols, colors and size. Here, we illustrate its utilities and capabilities in deriving biological insights from multi-omics dataset from GEO/Encode.</p>
<div id="refs" class="references">
<div id="ref-gao2013integrative">
<p>1. Gao, J. <em>et al.</em> Integrative analysis of complex cancer genomics and clinical profiles using the cBioPortal. <em>Science signaling</em> <strong>6,</strong> pl1 (2013).</p>
</div>
<div id="ref-finn2006pfam">
<p>2. Finn, R. D. <em>et al.</em> Pfam: Clans, web tools and services. <em>Nucleic acids research</em> <strong>34,</strong> D247–D251 (2006).</p>
</div>
<div id="ref-michael_p_schroeder_2015_14561">
<p>3. Schroeder, M. P. Muts-needle-plot: Mutations needle plot v0.8.0. (2015). doi:<a href="https://doi.org/10.5281/zenodo.14561">10.5281/zenodo.14561</a></p>
</div>
<div id="ref-bock2005biq">
<p>4. Bock, C. <em>et al.</em> BiQ analyzer: Visualization and quality control for dna methylation data from bisulfite sequencing. <em>Bioinformatics</em> <strong>21,</strong> 4067–4068 (2005).</p>
</div>
<div id="ref-mallona2014methylation">
<p>5. Mallona, I., Díez-Villanueva, A. &amp; Peinado, M. A. Methylation plotter: A web tool for dynamic visualization of dna methylation data. <em>Source code for biology and medicine</em> <strong>9,</strong> 1 (2014).</p>
</div>
<div id="ref-zackay2010methvisual">
<p>6. Zackay, A. &amp; Steinhoff, C. MethVisual-visualization and exploratory statistical analysis of dna methylation profiles from bisulfite sequencing. <em>BMC research notes</em> <strong>3,</strong> 337 (2010).</p>
</div>
<div id="ref-zhu4redseq">
<p>7. Zhu, L. &amp; others. REDseq: A bioconductor package for analyzing high throughput sequencing data from restriction enzyme digestion.</p>
</div>
<div id="ref-skidmore2016genvisr">
<p>8. Skidmore, Z. L. <em>et al.</em> GenVisR: Genomic visualizations in r. <em>Bioinformatics</em> btw325 (2016).</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sean Davis.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
